core/src/main/java/com/oath/oak/MinKeyWriteBuffer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+
+class MinKeyWriteBuffer extends MinKeyReadBuffer implements OakWriteBuffer, OakUnsafeDirectBuffer {
```

Comment:
Again if we use existing key, min key doesn't need to be written. Two create two additional classes just for that sounds to me over-engineering. Also that it looks to me not effective, if we need to change it from ByteBuffer, let's use existing keys.

Reply:
Modify to allocate min-key using Oak's allocator